lib/driver.rb

```diff
@@ -23,25 +22,55 @@ def initialize(input)
 
     def average_rating
       total_ratings = 0
-      @trips.each do |trip|
+      finish_trip.each do |trip|
         total_ratings += trip.rating
       end
 
-      if trips.length == 0
+      if finish_trip.length == 0
         average = 0
       else
-        average = (total_ratings.to_f) / trips.length
+        average = (total_ratings.to_f) / finish_trip.length
       end
 
       return average
     end
 
     def add_trip(trip)
-      if trip.class != Trip
+      unless trip.class <= Trip
         raise ArgumentError.new("Can only add trip instance to trip collection")
       end
 
+      if trip.end_time == nil
+        @status = :UNAVAILABLE
+      end
+
       @trips << trip
     end
-  end
-end
+
+    def total_revenue
+      fee = 1.65
+      driver_takehome = 0.8
+
+      subtotal = 0
+      finish_trip.each do |trip|
+          subtotal += trip.cost - fee
+      end
+
+      total = subtotal * driver_takehome
+      return total.round(2)
+    end
+
+    def average_revenue
+      average = 0
+      finish_trip.each do |trip|
+        average += (total_revenue / (trip.duration / 3600.0))
+      end
+      return average.round(2)
+    end
+
+    def finish_trip
+      trips.reject {|trip| trip.end_time == nil}
+    end
```

Comment:
I am however not sure I agree with the name of this method. `finish_trip` sounds to me like a command: "Hey driver, finish your (most recent?) trip". Instead you might name it `finished_trips`, which sounds like it's asking for information (and also implies it'll be giving back an array instead of a single object).